src/main/java/io/kontur/disasterninja/service/ApplicationService.java

```diff
@@ -0,0 +1,37 @@
+package io.kontur.disasterninja.service;
+
+import io.kontur.disasterninja.client.UserProfileClient;
+import io.kontur.disasterninja.dto.AppDto;
+import io.kontur.disasterninja.dto.FeatureDto;
+import io.kontur.disasterninja.util.AuthenticationUtil;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class ApplicationService {
+
+    private final UserProfileClient userProfileClient;
+
+    public AppDto getAppConfig(UUID appId) {
+        if (appId == null) {
+            appId = UUID.fromString(Objects.requireNonNull(userProfileClient.getDefaultAppId().getBody()));
```

Comment:
I would define default app id in UPS itself. Thus this roundtrip can be removed.